lib/topological_inventory/ingress_api/messaging_client.rb

```diff
@@ -0,0 +1,38 @@
+module TopologicalInventory
+  module IngressApi
+    def self.with_messaging_client
+      messaging_client ||= raw_messaging_client
+      raise if messaging_client.nil?
+
+      begin
+        yield messaging_client
+      rescue Kafka::MessageSizeTooLarge
+        # Don't reset the connection for user-error
+        raise
+      rescue Kafka::Error
+        # If we hit an underlying kafka error then reset the connection
+        messaging_client.close
+        self.messaging_client = nil
+
+        raise
+      end
+    end
+
+    private_class_method def self.messaging_client
+      Thread.current[:messaging_client]
+    end
+
+    private_class_method def self.messaging_client=(value)
+      Thread.current[:messaging_client] = value
+    end
+
+    private_class_method def self.raw_messaging_client
```

Comment:
Maybe call this `new_messaging_client`? I feel like that's more clear from the caller that we're initializing something rather than getting access to some kind of underlying implementation detail (which is what I think when I see `raw_`)

e.g.
```ruby
irb(main):001:0> ApplicationRecord.connection.raw_connection
=> #<PG::Connection:0x0000563bdf3a2f70>

```

Reply:
:+1: done